OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -263,6 +274,9 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 				FireDelay = Util.ApplyPercentageModifiers(Weapon.ReloadDelay, modifiers);
 				Burst = Weapon.Burst;
 
+				if (Weapon.ReloadSound != null && Weapon.ReloadSound.Any())
```

Comment:
Use `ScheduleDelayedAction` here to avoid duplicating the boilerplate for ticking down and checking instant vs delayed playback.